API.md

```diff
@@ -1,64 +1,272 @@
+
+```javascript
+{
+  json: 'boolean',         //Whether this will be a JSON body. Overrides value set in globalSetup().
+  body: 'string'|'object', //The body to include in the outbound request. This overrides "data" if provided in method call.
+  mock: 'function',        //A mock runner to use. When not provided, uses "request" (i.e. does it for real).
+  form: 'boolean'          //Use the object in the body to create a form-encoded request.
+}
+```
+
+### get(uri, params)
+
+- Types: `uri`: `string`, `params`: `object` (optional)
+- Default: `none`
+
+Perform an HTTP GET on the specified URI.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com/login')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### head(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP HEAD request.
+
+### options(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP OPTIONS request.
+
+### post(uri, data, params)
+
+- Types: `uri`: `string`, `data`:`object` `params`: `object` (optional)
+- Default: `none`
+
+Perform an HTTP POST to the specified URI.
+
+```javascript
+frisby.create('a test')
+    .post('http://example.com/login', {
+      username: 'joe@example.com',
+      password: 'J0£_£x@mpl£'
+    })
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+frisby.create('another test')
+    .post('http://example.com/contact-us', {
+      name: 'Test Person',
+      comment: 'What a lovely test'
+    }, {form: true})
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+```
+
+### put(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PUT request.
+
+### patch(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PATCH request.
+
+### delete(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP DELETE request.
+
+### addHeader(header, content)
+
+Adds an HTTP header to your request
+
+- Types: `header`: `string`, `content`: `string`
+- Default: `none`
+
+Note that `content` is always a string, regardless of whether the data it would represent is something else (an integer or GUID for example).
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .addHeader('Accept', 'text/html')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### addHeaders(headers)
+
+Adds a collection of headers to your request
+
+- Types: `headers`: `object`
+- Default: `none`
+
+A flat object where each key becomes a header name and each corresponding value becomes that header's value in the request.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .addHeaders({
+      accept: 'text/html',
+      referer: 'http://www.test.net'
```

Comment:
You could mention in a comment that the header field names are case insensitive, according to the HTTP spec?

Reply:
Done.